rm_reporting/controllers/case_controller.py

```diff
@@ -25,8 +27,13 @@ def get_case_data(collection_exercise_id: str) -> list:
         "ORDER BY sample_unit_ref, status"
     )
 
-    with case_engine.begin() as conn:
-        return conn.execute(case_business_ids_query, {"collection_exercise_id": collection_exercise_id}).all()
+    try:
+        with case_engine.begin() as conn:
+            result = conn.execute(case_business_ids_query, {"collection_exercise_id": collection_exercise_id}).all()
+    except SQLAlchemyError:
+        logger.error("Failed to get case data", collection_exercise_id=collection_exercise_id)
+        abort(404, "Case data not found")
+    return result
```

Comment:
My bad, I was moving things around and I neglected to make the code consistent in different places. I have now changed the code so that the `return {VARIABLE}` statement in the block near line 70-ish is moved outside the `try:` block to make it consistent with the pattern used by the one at the bottom of the file, as well as the functions in `party_controller.py` (which were like that prior to me making any changes as part of this PR).